flight_computer/src/flash/reader.cpp

```diff
@@ -250,48 +254,48 @@ void parse_recording(uint16_t flight_num, rec_entry_type_e filter_mask) {
-          size_t elem_sz = sizeof(rec_elem.u.filtered_data_info);
-          lfs_file_read(&lfs, &curr_file, (uint8_t *)&rec_elem.u.imu, elem_sz);
+          const size_t elem_sz = sizeof(rec_elem.u.filtered_data_info);
+          lfs_file_read(&lfs, &curr_file, reinterpret_cast<uint8_t *>(&rec_elem.u.imu), elem_sz);
           if ((rec_type_without_id & filter_mask) > 0) {
             log_raw("%lu|FILTERED_DATA_INFO|%f|%f", rec_elem.ts,
-                    (double)rec_elem.u.filtered_data_info.filtered_altitude_AGL,
-                    (double)rec_elem.u.filtered_data_info.filtered_acceleration);
+                    static_cast<double>(rec_elem.u.filtered_data_info.filtered_altitude_AGL),
+                    static_cast<double>(rec_elem.u.filtered_data_info.filtered_acceleration));
           }
         } break;
         case FLIGHT_STATE: {
-          size_t elem_sz = sizeof(rec_elem.u.flight_state);
-          lfs_file_read(&lfs, &curr_file, (uint8_t *)&rec_elem.u.imu, elem_sz);
+          const size_t elem_sz = sizeof(rec_elem.u.flight_state);
+          lfs_file_read(&lfs, &curr_file, reinterpret_cast<uint8_t *>(&rec_elem.u.imu), elem_sz);
           if ((rec_type_without_id & filter_mask) > 0) {
             log_raw("%lu|FLIGHT_STATE|%s", rec_elem.ts, GetStr(rec_elem.u.flight_state, fsm_map));
           }
         } break;
         case EVENT_INFO: {
-          size_t elem_sz = sizeof(rec_elem.u.event_info);
-          lfs_file_read(&lfs, &curr_file, (uint8_t *)&rec_elem.u.imu, elem_sz);
+          const size_t elem_sz = sizeof(rec_elem.u.event_info);
+          lfs_file_read(&lfs, &curr_file, reinterpret_cast<uint8_t *>(&rec_elem.u.imu), elem_sz);
           if ((rec_type_without_id & filter_mask) > 0) {
-            peripheral_act_t action = rec_elem.u.event_info.action;
+            const peripheral_act_t action = rec_elem.u.event_info.action;
             log_raw("%lu|EVENT_INFO|%s|%s|%d", rec_elem.ts, GetStr(rec_elem.u.event_info.event, event_map),
                     GetStr(rec_elem.u.event_info.action.action, action_map), action.action_arg);
           }
         } break;
         case ERROR_INFO: {
-          size_t elem_sz = sizeof(rec_elem.u.error_info);
-          lfs_file_read(&lfs, &curr_file, (uint8_t *)&rec_elem.u.imu, elem_sz);
+          const size_t elem_sz = sizeof(rec_elem.u.error_info);
+          lfs_file_read(&lfs, &curr_file, reinterpret_cast<uint8_t *>(&rec_elem.u.imu), elem_sz);
           if ((rec_type_without_id & filter_mask) > 0) {
             log_raw("%lu|ERROR_INFO|%lu", rec_elem.ts, rec_elem.u.error_info.error);
           }
         } break;
         case GNSS_INFO: {
-          size_t elem_sz = sizeof(rec_elem.u.gnss_info);
-          lfs_file_read(&lfs, &curr_file, (uint8_t *)&rec_elem.u.imu, elem_sz);
+          const size_t elem_sz = sizeof(rec_elem.u.gnss_info);
+          lfs_file_read(&lfs, &curr_file, reinterpret_cast<uint8_t *>(&rec_elem.u.imu), elem_sz);
```

Comment:
Were those reinterpret_cast tested? Can you read from the file system?

Reply:
Yes. In general reinterpret_cast doesn't do anything fancy, it's more or less the same as doing a C-style cast, you only declare your intent with this. In this case, we are reinterpreting a pointer to an IMU object to a pointer to `uint8_t`, which we were already doing.